test_runner/src/main/kotlin/ftl/reports/api/CreateJUnitTestResult.kt

```diff
@@ -7,6 +7,7 @@ internal fun List<TestExecution>.createJUnitTestResult(
     withStackTraces: Boolean = false
 ) = JUnitTestResult(
     testsuites = this
+        .filterNullToolResultsStep()
```

Comment:
1. I wonder why FTL is returning `null` for `toolResultsStep`? 
2. Can we reproduce this error? 
3. Its safe to skip some tests execution and get correct results?
4. Maybe flank should display proper warning about some unexpected behaviour?

Reply:
AD 1. If matrix is invalid FTL returns `null` as `toolResultsStep`
AD 2. This is more to avoid crashing flank in the future, so far, for now, I do not know how to reproduce this issue
AD 3. This is safe because tests results are based on a request made with `toolResultsStep` which will not succeed without `toolResultsStep` 
AD 4. Ideally, we should not trigger INVALID matrix request, but for now, we do not know which settings will cause matrix invalid (see point 2)

> 1. I wonder why FTL is returning `null` for `toolResultsStep`?
> 2. Can we reproduce this error?
> 3. Its safe to skip some tests execution and get correct results?
> 4. Maybe flank should display proper warning about some unexpected behaviour?

AD 1. If matrix is invalid FTL returns `null` as `toolResultsStep`
AD 2. This is more to avoid crashing flank in the future, so far, for now, I do not know how to reproduce this issue
AD 3. This is safe because tests results are based on a request made with `toolResultsStep` which will not succeed without `toolResultsStep` 
AD 4. Ideally, we should not trigger INVALID matrix request, but for now, we do not know which settings will cause matrix invalid (see point 2)